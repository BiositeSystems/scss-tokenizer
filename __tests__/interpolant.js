var scss = require('..');
var fs = require('fs');
var path = require('path');

var fixture = function(name) {
    return fs.readFileSync(
        path.join(__dirname, 'fixture', name)
    );
}

describe('interpolant', function() {
    it('should tokenize an interpolant', function() {
        const tree = scss.tokenize(fixture('interpolant.scss'));
        expect(tree).toMatchSnapshot();
    });
});
