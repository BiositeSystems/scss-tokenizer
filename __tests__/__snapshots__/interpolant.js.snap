// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interpolant should tokenize an interpolant 1`] = `
Array [
  Array [
    "*",
    "*",
    1,
    1,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "{",
    "{",
    1,
    3,
  ],
  Array [
    "newline",
    "
",
    2,
    0,
  ],
  Array [
    "space",
    "	",
  ],
  Array [
    "ident",
    "background-image",
    2,
    2,
    2,
    17,
  ],
  Array [
    ":",
    ":",
    2,
    18,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "ident",
    "url",
    2,
    20,
    2,
    22,
  ],
  Array [
    "(",
    "(",
    2,
    23,
  ],
  Array [
    "\\"",
    "\\"",
    2,
    24,
  ],
  Array [
    "startInterpolant",
    "#{",
    2,
    30,
  ],
  Array [
    "$",
    "$",
    2,
    31,
  ],
  Array [
    "ident",
    "file-1x",
    2,
    32,
    2,
    38,
  ],
  Array [
    "endInterpolant",
    "}",
    2,
    39,
  ],
  Array [
    "\\"",
    "\\"",
    2,
    40,
  ],
  Array [
    ")",
    ")",
    2,
    37,
  ],
  Array [
    ";",
    ";",
    2,
    38,
  ],
  Array [
    "newline",
    "
",
    3,
    0,
  ],
  Array [
    "}",
    "}",
    3,
    1,
  ],
  Array [
    "newline",
    "
",
    4,
    0,
  ],
]
`;
